xtras/src/spawner.rs

```diff
@@ -0,0 +1,128 @@
+use crate::SendAsyncSafe;
+use async_trait::async_trait;
+use futures::Future;
+use tokio_tasks::TaskMap;
+use uuid::Uuid;
+use xtra::Context;
+use xtra_productivity::xtra_productivity;
+
+/// A spawner actor controls the lifetime of tasks it is instructed to
+/// spawn. If all (strong) `Address`es to this `xtra::Actor` are
+/// dropped, all the tasks it is tracking will be cancelled.
+///
+/// Furthermore, a spawner actor will learn about any spawned task
+/// that has finished, dropping the corresponding task handle and
+/// freeing up memory.
+pub struct Actor {
```

Comment:
I think that his pattern might actually be good enough - not sure we need the `TaskMap` if we have this. 
I imagine that we can have multiple of this `spawner::Actor`, we could just scale it up with multiple actors. 
What do you think? Is it worth to have both?

Reply:
How do you see this actor working without the underlying `TaskMap`?

Reply:
Don't know what I was thinking when I wrote this. Somehow I thought we are using `TaskMap` directly in the actor, but what we are doing *is* using this `Actor` to manage tasks, and `TaskMap` is only used by this actor. 
Sorry for the confusion, you can ignore my comment!